src/components/VCheckbox.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <label :for="id" class="checkbox-label" :class="labelClasses">
+    <input
+      :id="id"
+      type="checkbox"
+      class="checkbox"
+      v-bind="inputAttrs"
+      @change="onChange"
+    />
+    <Checkmark
+      v-show="localCheckedState"
+      class="checkmark"
+      focusable="false"
+      width="20"
+      height="20"
+      role="img"
+    />
+    <!--  @slot The checkbox label  --><slot />
+  </label>
+</template>
+
+<script>
+import Checkmark from '~/assets/icons/checkmark.svg?inline'
+import { computed, defineComponent, ref, watch } from '@nuxtjs/composition-api'
+
+const VCheckbox = defineComponent({
+  name: 'VCheckbox',
+  components: { Checkmark },
+  props: {
+    /**
+     * Checkbox `id` is used for the input id property, connecting the label to
+     * the checkbox. id is also used in the `change` event payload as the `name`
+     * and `value` parameters if they are not set.
+     */
+    id: {
```

Comment:
I'm not 100% sure but since ID is an HTML attribute, this might conflict. For disambiguation, I used `fieldId` in the `InputField` component.

Reply:
This `id` prop is used both for the HTML attribute (mainly to get the `<label for="id">` working well, since I've read that simply wrapping an input in a label might not be sufficient for some screen readers), and to use as a fall-back for `name` and/or `value` properties. I can see that a non-unique id might cause a problem, but don't know if it can be significant. Can it?